controllers/podchaos/containerkill/types.go

```diff
@@ -0,0 +1,187 @@
+	if err = r.Get(ctx, req.NamespacedName, &podchaos); err != nil {
+		r.Log.Error(err, "unable to get podchaos")
+		return ctrl.Result{}, nil
+	}
+
+	if podchaos.Spec.ContainerName == "" {
+		err = errors.New("the ContainerName is empty")
+		r.Log.Error(err, "the ContainerName is empty")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	shouldAct := podchaos.GetNextStart().Before(now)
+	if !shouldAct {
+		return ctrl.Result{RequeueAfter: podchaos.GetNextStart().Sub(now)}, nil
+	}
+	pods, err := utils.SelectPods(ctx, r.Client, podchaos.Spec.Selector)
+	if err != nil {
+		r.Log.Error(err, "fail to get selected pods")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if len(pods) == 0 {
+		err = errors.New("no pod is selected")
+		r.Log.Error(err, "no pod is selected")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	filteredPod, err := utils.GeneratePods(pods, podchaos.Spec.Mode, podchaos.Spec.Value)
+	if err != nil {
+		r.Log.Error(err, "fail to generate pods")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	g := errgroup.Group{}
+	haveSelected := false
+	haveKilled := false
+	for index := range filteredPod {
+		pod := &filteredPod[index]
+
+		//for index := range pod.Spec.Containers {
+		for index := range pod.Status.ContainerStatuses {
+			containerName := pod.Status.ContainerStatuses[index].Name
+			containerID := pod.Status.ContainerStatuses[index].ContainerID
+			//container := pod.Spec.Containers[index]
+
+			if containerName == podchaos.Spec.ContainerName {
+				haveSelected = true
+				err = r.KillContainer(ctx, pod, containerID)
+				if err != nil {
+					r.Log.Error(err, "failed to kill container")
+				} else {
+					haveKilled = true
+				}
+			}
+		}
+	}
+
+	if haveSelected == false {
+		err = errors.New("no container is selected")
+		r.Log.Error(err, "no container is selected")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if haveKilled == false {
+		err = errors.New("no container is killed")
+		r.Log.Error(err, "no container is killed")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if err := g.Wait(); err != nil {
+		return ctrl.Result{}, nil
+	}
+	next, err := utils.NextTime(*podchaos.Spec.Scheduler, now)
+	if err != nil {
+		r.Log.Error(err, "failed to get next time")
+		return ctrl.Result{}, nil
+	}
+
+	podchaos.SetNextStart(*next)
+
+	podchaos.Status.Experiment.StartTime = &metav1.Time{
+		Time: now,
+	}
+	podchaos.Status.Experiment.EndTime = &metav1.Time{
+		Time: now,
+	}
+
+	podchaos.Status.Experiment.Pods = []v1alpha1.PodStatus{}
+	for _, pod := range pods {
+		ps := v1alpha1.PodStatus{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+			HostIP:    pod.Status.HostIP,
+			PodIP:     pod.Status.PodIP,
+			Action:    string(podchaos.Spec.Action),
+			Message:   containerKillActionMsg,
+		}
+
+		podchaos.Status.Experiment.Pods = append(podchaos.Status.Experiment.Pods, ps)
+	}
+	if err := r.Update(ctx, &podchaos); err != nil {
+		r.Log.Error(err, "unable to update chaosctl status")
```

Comment:
Hmm, may I ask what does this `chaosctl` mean

Reply:
Actually, I see this "log msg" appear multiple times in the project. So this is  consistent with other places.

Reply:
It is not clear enough to users.